src/h3lib/lib/h3Index.c

```diff
@@ -314,7 +314,15 @@ int H3_EXPORT(compact)(const H3Index* h3Set, H3Index* compactedSet,
                         hashSetArray[loc] & H3_RESERVED_MASK_NEGATIVE;
                     if (tempIndex == parent) {
                         int count = H3_GET_RESERVED_BITS(hashSetArray[loc]) + 1;
-                        if (count > 7) {
+                        int limitCount = 7;
+                        if (H3_EXPORT(h3IsPentagon)(
```

Comment:
I suspect this will have a negative impact on the performance of `compact`. I wonder how much, though.